helion/_compiler/device_ir.py

```diff
@@ -211,22 +211,23 @@ class DeviceIR:
     def __init__(self) -> None:
         super().__init__()
         self.graphs: list[GraphInfo] = []
-        self.root_id: int | None = None
+        self.root_ids: list[int] = []
         self.rolled_reductions: list[RolledReductionInfo] = []
         self.grid_block_indices: list[list[int]] = []
 
-    def get_root(self, config: Config) -> torch.fx.GraphModule:
+    def get_root(self, config: Config, root_id: int) -> torch.fx.GraphModule:
         """ " If we are using a rolled reduction, return the rolled reduction graph otherwise
         return the root graph."""
-        if (root_id := self.root_id) is None:
-            raise AssertionError("No root graph")
+        if root_id >= len(self.root_ids):
+            raise AssertionError("Invalid root graph")
+        rid = self.root_ids[root_id]
```

Comment:
Is this extra layer of indirection actually needed?  Could just use the graph ID directly?